.travis.yml

```diff
@@ -101,210 +76,197 @@ _helpers:
     on:
       tags: true
       all_branches: true
-- &osx_pypi_deploy_base_1011
-  <<: *osx_python_base
-  <<: *generic_deploy_base
-  osx_image: xcode7.3
-  script: skip
-  after_success: []
+
+- &_osx_pypi_deploy_base
+  <<: *_generic_deploy_base
+  stage: &stage_test_osx_name test under OS X (last chance to fail before deploy available)
+  os: osx
+  language: shell
   env:
-  - &env_os1011_msg Build and deploy to PYPI of OS X 10.11 binary wheel=
-  - &env_py36 PYTHON_VERSION=3.6.3
-  - *env_pyenv
-  - *env_path
+  - &env_pyenv PYENV_ROOT="$HOME/.pyenv"
+  - &env_path PATH="$PYENV_ROOT/bin:$PATH"
+  before_install:
+  - brew update
+  - |
```

Comment:
1) This all is meant to be in `_helpers`. I put it there for a reason: big chunks of logic in the middle of matrix reduce readability.
2) If you want to help with this, please port https://github.com/amol-/dukpy/blob/69f56f3/.travis.yml#L41-L120 here, it's a speed-up for tests. But keep a brew implementation: it's needed for dev/nightly.
3) This change is too big and will need more adjustments and closer look. So please factor it out into a separate PR. Otherwise it'll be hard to review and reason about this which will delay merging and block even smaller improvements.

Reply:
This *is* in helpers.

Otherwise, happy to defer the pyenv / Python.org discussion. I did address this in the PR discussion, and agree that using a downloaded Python is a much better idea.

Reply:
Ah.. I must've missed that with huge chunks moved around.